homeassistant/components/amazon_polly/tts.py

```diff
@@ -202,39 +124,48 @@ def get_engine(hass, config, discovery_info=None):
 class AmazonPollyProvider(Provider):
```

Comment:
It for lines starting from 110, github doesn't allow to add comments outside if edited context.

1. Define type for empty collections.
2. Define type for `Any` when possible.
3. Handle `None`.
```python
    supported_languages: list[str] = []

    all_voices: dict[str, dict[str, str]] = {}

    all_voices_req = polly_client.describe_voices()

    for voice in all_voices_req.get("Voices", []):
        voice_id: str | None = voice.get("Id")
        if voice_id is None:
            continue
        all_voices[voice_id] = voice
        language_code: str | None = voice.get("LanguageCode")
        if language_code is not None and language_code not in supported_languages:
            supported_languages.append(language_code)
```